taxonomies/additives.txt

```diff
@@ -3025,39 +3066,86 @@ organic_eu:en:authorized
-de:E171, Titandioxid, Titan(IV)-oxid, TiO2, E 171, Titanweiss, Rutilweiß
+cs:E171, Oxid titaničitý, TiO2, kysličník titaničitý
+da:E171, Titandioxid, Titanoxid
+de:E171, Titandioxid, Titan(IV)-oxid, TiO2, Titanweiss, Rutilweiß
 el:E171, Διοξειδιο του τιτανιου
+eo:E171, titana dioksido
 es:E171, Dióxido de titanio, Óxido de titanio, Oxido de titanio(IV)
 et:E171, Titaandioksiid
-fi:E171, Titaanidioksidi, Titaanivalkoinen, E 171, Titaani(IV)oksidi, Titaanioksidi, Tio2
-fr:E171, Oxyde de titane, dioxyde de titane, bioxyde de titane, CI 77891, CI Pigment White 6, CAS 13463-67-7, TiO2, 13463-67-7
+fa:E171, تیتانیوم دی اکسید
+fi:E171, Titaanidioksidi, Titaanivalkoinen, Titaani(IV)oksidi, Titaanioksidi, Tio2
+fr:E171, Oxyde de titane, dioxyde de titane, bioxyde de titane, CI Pigment White 6
+ga:E171, dé-ocsaíd tíotáiniam
+he:E171, טיטניום דו-חמצני
+hi:E171, टाइटानियम डाइऑक्साइड
 hu:E171, Titán-dioxid
+hy:E171, Տիտանի երկօքսիդ
+id:E171, Titanium dioksida
+is:E171, Títandíoxíð
 it:E171, Biossido di titanio, Diossido di titanio, TiO2
+ja:E171, 酸化チタン
+ko:E171, 이산화 타이타늄
+ky:E171, Титан оксиддери
 lt:E171, Titano dioksidas, Titano baltalai
 lv:E171, Titāna dioksīds
+ml:E171, ടൈറ്റാനിയം ഡൈഓക്സൈഡ്
+ms:E171, Titanium dioksida
 mt:E171, Diossidu tat-titanju
+nb:E171, titandioxid
 nl:E171, Titaandioxide, Titanium(IV)oxide, Titaniumoxide, Titaniumdioxide, Titanium dioxide, Titaanwit
 pl:E171, Dwutlenek tytanu, tlenek tytanu(IV), ditlenek tytanu, biel tytanowa
 pt:E171, Dióxido de titânio, Óxido de titânio
 ro:E171, Dioxid de titan, Bioxid de titan
-ru:E171, Титана диоксид, оксид титана(IV), Е171, TiO2, двуокись титана, оксид титана, белила титановые, титановые белила, диоксид титана
+ru:E171, Титана диоксид, оксид титана(IV), TiO2, двуокись титана, оксид титана, белила титановые, титановые белила, диоксид титана
+sh:E171, Titanijum dioksid
+si:E171, ටයිටේනියම් ඩයොක්සයිඩ්
 sk:E171, Oxid titaničitý, Titánová beloba
 sl:E171, Titanov dioksid
-sv:E171, Titandioxid, Titaniumdioxid, TiO2, E 171, Titansyra
+sr:E171, титанијум диоксид
+sv:E171, Titandioxid, Titaniumdioxid, TiO2, Titansyra
+ta:E171, தைட்டானியம் ஈராக்சைடு
+tr:E171, Titanyum dioksit
+uk:E171, Білила титанові
+vi:E171, Titani dioxide
+zh:E171, 二氧化鈦
+# azb:E171, تیتانیوم دی اوکسید
+# sco:E171, titanium dioxide
+# sr-ec:E171, титанијум диоксид
+# sr-el:E171, titanijum dioksid
+# war:E171, Titanium dioxide
+# wuu:E171, 二氧化钛
+# zh-cn:E171, 二氧化钛
+# zh-hans:E171, 二氧化钛
+# zh-hant:E171, 二氧化鈦
+# zh-hk:E171, 二氧化鈦
+# zh-sg:E171, 二氧化钛
+# zh-tw:E171, 二氧化鈦
 e_number:en:171
 wikidata:en:Q193521
+wikipedia:en:https://en.wikipedia.org/wiki/Titanium_dioxide
+# additive/e171-titanium-dioxide has 4518 products in 15 languages @2019-03-27
 efsa_evaluation_url:en: http://dx.doi.org/10.2903/j.efsa.2016.4545
 efsa_evaluation_date:en: 2016-09-14
 efsa_evaluation:en: Re-evaluation of titanium dioxide (E 171) as a food additive
 additives_classes:en: en:colour
 vegan:en:yes
 vegetarian:en:yes
 
+<en:E171
+en:colorant E171
```

Comment:
This should be removed, it will be handled by the parser.

Reply:
Some additives have multiple purposes. Encoding it like this emphasises why the producer added an additive to the product. I do not know if you keep this information in the parser.

In addition it also helps if you use the taxonomy for translation, as this entry can be used for that.

Reply:
Producers have to indicate the additive class before the additive, so there should always be something like coloring: E171, E102 and E120. During parsing, we keep track of the additive class, so that we can associate it not only with E171, but also the additives after.